front/components/assistant_builder/AssistantBuilderTablesModal.tsx

```diff
@@ -131,18 +175,22 @@ export default function AssistantBuilderTablesModal({
         tableName: table.name,
       }));
 
-      onSave(configs, selectedDataSource);
+      onSave(configs, selectedDataSourceView);
     } finally {
       setIsSaving(false);
     }
   }
 
-  const onClose = () => {
-    setOpen(false);
-    setTimeout(() => {
-      setSelectedDataSource(null);
-      setIsSaving(false);
-    }, 200);
+  const onClose = (e?: Event, forceClose: boolean = false) => {
```

Comment:
Why do we need the event here?

Reply:
It's sent by the component itself.
We don't _need_ it but we are interested in the second argument.

Reply:
I see except that it's not consistent in `Modal`. Anyway something for later.